contract/contracts/services/DC.sol

```diff
@@ -0,0 +1,222 @@
+        IRegistrarController.Price memory price = registrarController.rentPrice(name, duration);
+        return price.base + price.premium;
+    }
+
+    /**
+     * @dev `register` calls RegistrarController register and is used to register a name
+     * this also takes a fee for the web2 registration which is held by DC.sol a check is made to ensure the value sent is sufficient for both fees
+     * @param name The name to be registered e.g. for test.country it would be test
+     # @param owner The owner of the registerd name
+     * @param duration Length of time to register the name
+     * @param secret A random secret passed by the client
+     */
+    function register(string calldata name, address owner, uint256 duration, bytes32 secret) public payable whenNotPaused {
+        require(bytes(name).length <= 128, "DC: name too long");
+        uint256 price = getENSPrice(name, duration);
+        require(price <= msg.value, "DC: insufficient payment");
+        require(available(name), "DC: name unavailable");
+        _register(name, owner, duration, secret);
+
+        // Return any excess funds
+        uint256 excess = msg.value - price;
+        if (excess > 0) {
+            (bool success, ) = msg.sender.call{value: excess}("");
+            require(success, "cannot refund excess");
+        }
+    }
+
+    /**
+     * @dev `_register` calls RegistrarController register and is used to register a name
+     * it is passed a value to cover the costs of the ens registration
+     * @param name The name to be registered e.g. for test.country it would be test
+     * @param owner The owner address of the name to be registered
+     * @param duration Length of time to register the name
+     * @param secret A random secret passed by the client
+     */
+    function _register(string calldata name, address owner, uint256 duration, bytes32 secret) internal whenNotPaused {
+        uint256 ensPrice = getENSPrice(name, duration);
+        bytes[] memory emptyData;
+        registrarController.register{value: ensPrice}(name, owner, duration, secret, resolver, emptyData, reverseRecord, fuses, wrapperExpiry);
+    }
+
+    /**
+     * @dev `renew` calls RegistrarController renew and is used to renew a name
+     * @param name The name to be registered e.g. for test.country it would be test
+     * @param duration Length of time to register the name
+     */
+    function renew(string calldata name, uint256 duration) public payable whenNotPaused {
+        uint256 price = getENSPrice(name, duration);
+        require(price <= msg.value, "DC: insufficient payment");
+        registrarController.renew{value: price}(name, duration);
+        uint256 excess = msg.value - price;
+        if (excess > 0) {
+            (bool success, ) = msg.sender.call{value: excess}("");
+            require(success, "cannot refund excess");
+        }
+    }
+
+    function verifyOnwer(string calldata name, address owner) public view returns(bool) {
```

Comment:
this and other owner view-functions should auto-detect and support both NameWrapper and BaseRegistrar no matter the wrap status of the domain.